### PR TITLE
Implemented cross-cluster monitor support

### PR DESCRIPTION
**Pull request by AWSHurneyt**
### Description
Added `clusters` field to support cross-cluster cluster metrics monitors.

Documentation PR for reference https://github.com/opensearch-project/documentation-website/pull/6350
 
### Issues Resolved
https://github.com/opensearch-project/alerting-dashboards-plugin/issues/796
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/kotlin/org/opensearch/commons/alerting/model/Alert.kt [hunk 1]
```diff
@@ -43,6 +43,7 @@ data class Alert(
     val aggregationResultBucket: AggregationResultBucket? = null,
     val executionId: String? = null,
     val associatedAlertIds: List<String>,
+    val clusters: List<String>? = null,
```
Comment (eirsep) [inline src/main/kotlin/org/opensearch/commons/alerting/model/Alert.kt:46]: will this break security analytics constructor?? if yes, plz create PR for that too

Comment (eirsep) [reply src/main/kotlin/org/opensearch/commons/alerting/model/Alert.kt:46]: what is stored here? can you share an example list of opensource clusters?

Comment (eirsep) [inline src/main/kotlin/org/opensearch/commons/alerting/model/Alert.kt:46]: plz udpate the alert parsing unit tests to account for this field

**Review by eirsep — CHANGES_REQUESTED** [review]

File: src/main/kotlin/org/opensearch/commons/alerting/model/Alert.kt (hunk 1 above)
Comment (eirsep) [reply src/main/kotlin/org/opensearch/commons/alerting/model/Alert.kt:46]: should this be called clusterNames or clusterIds

File: src/main/kotlin/org/opensearch/commons/alerting/model/ClusterMetricsInput.kt [hunk 2]
```diff
@@ -74,6 +75,7 @@ data class ClusterMetricsInput(
             .field(PATH_FIELD, path)
             .field(PATH_PARAMS_FIELD, pathParams)
             .field(URL_FIELD, url)
+            .field(CLUSTERS_FIELD, clusters)
```
Comment (eirsep) [inline src/main/kotlin/org/opensearch/commons/alerting/model/ClusterMetricsInput.kt:78]: same.. update parsing tests for clusterMetricsInput model or plz add a test if non-existent

**Review by eirsep — CHANGES_REQUESTED** [review]

File: src/main/kotlin/org/opensearch/commons/alerting/model/Alert.kt (hunk 1 above)
Comment (AWSHurneyt) [reply src/main/kotlin/org/opensearch/commons/alerting/model/Alert.kt:46]: Created this PR to update SAP https://github.com/opensearch-project/security-analytics/pull/837

File: src/main/kotlin/org/opensearch/commons/alerting/model/ClusterMetricsInput.kt (hunk 2 above)
Comment (AWSHurneyt) [reply src/main/kotlin/org/opensearch/commons/alerting/model/ClusterMetricsInput.kt:78]: Revised.

**Review by praveensameneni — APPROVED** [review]

**Review by eirsep — APPROVED** [review]

**Review by sbcd90 — APPROVED** [review]

**Pull request merged**